### DISCUSSION TITLE
Change docker-compose deployment to curl-ing a .yaml file, not cloning the repo?

### DISCUSSION THREAD

**Discussion by sfirke** [apache/superset#24987] [Ideas]
Currently the [Superset docs tell a user who wants to try out a version of Superset to clone the repository, switch to a specific tag, and run docker-compose up](https://superset.apache.org/docs/installation/installing-superset-using-docker-compose/).  This involves knowing git and downloading the entire codebase, and new users are often tripped up.

Contrast that with [how Apache Airflow tells users to get started via Docker](https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html#fetching-docker-compose-yaml).  There they say, curl this .yaml file, make an .env file, and run docker compose up.

I think Superset would be better off pointing new users to this workflow.  It would also largely resolve the [problem of the project currently directing new users to the unstable nightly build instead of latest stable release](https://github.com/apache/superset/discussions/22244#discussioncomment-4347110).

We could keep the `git clone` instructions for people who want to build their own images & develop Superset.  Perhaps that could be moved off into its own page, like [Airflow's Contributor Quick Start guide](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst).

What do folks think?  If there is support for this I can create a PR.

**Comment by sebastianliebscher** [comment]
Yes, I think a single `curl docker-compose-non-dev.yml` should definitely be the default way to deploy Apache Superset via docker-compose :+1:

**Reply by sebastianliebscher** [reply]
@sfirke here is a PR draft https://github.com/apache/superset/pull/25546

**Comment by jspaezp** [comment]
Hello Everyone!

I am a new user learning how to get started with superset and it would have been a lot better to learn what is relevant!

Let me know if you could use some more insights on "someone new just getting started from the docs" :)
